src/core/model/types/numeric_type.h

```diff
@@ -29,8 +35,89 @@ class INumericType : public IMetrizableType {
 
     [[nodiscard]] virtual std::byte const* Min() const = 0;
     [[nodiscard]] virtual std::byte const* Max() const = 0;
+
+private:
+    inline void IntCastTo_(std::byte* value, TypeId to_type) const;
+    inline void DoubleCastTo_(std::byte* value, TypeId to_type) const;
 };
 
+template <typename T>
+T INumericType::GetValueAs(std::byte const* value) const {
+    switch (GetTypeId()) {
+        case TypeId::kDouble: {
+            model::Double data = INumericType::GetValue<Double>(value);
+            return static_cast<T>(data);
+        }
+        case TypeId::kInt: {
+            model::Int data = INumericType::GetValue<Int>(value);
+            return static_cast<T>(data);
+        }
+        default:
+            throw std::logic_error("undefined type conversion");
```

Comment:
as I mentioned in the previous review, `assert(false)` is required here since passing an object of incorrect type to this function is a desbordante programmer mistake